types/rdfjs__namespace/index.d.ts

```diff
@@ -0,0 +1,22 @@
+// Type definitions for @rdfjs/namespace 1.1
+// Project: https://github.com/rdfjs-base/namespace
+// Definitions by: Chris Wilkinson <https://github.com/thewilkybarkid>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { DataFactory, NamedNode } from 'rdf-js';
+
+declare function namespace(baseIRI: string, options?: namespace.BuilderOptions): namespace.NamespaceBuilder;
+
+declare namespace namespace {
+    interface NamespaceBuilder {
+        (property: TemplateStringsArray | string): NamedNode;
+
+        readonly [property: string]: NamedNode;
```

Comment:
My IDE (PhpStorm) complains if using the property `name` (confuses it with `Function.name`, so thinks it's a `string` instead).

TypeScript and eslint don't have the problem though; not sure if there's a way to keep both happy.

Reply:
Adding an explicit `name` property gets round it, but  should be unnecessary.

Reply:
My eslint setup complains about that `name` property. Here's another solution

```
interface NamespaceBuilderFunc {
  (property: TemplateStringsArray | string): NamedNode;
}
type NamespaceBuilder = Record<string, NamedNode> & Omit<NamespaceBuilderFunc, 'name'>
```

Reply:
Hmm, doesn't seem to work, no longer recognised as callable...

Reply:
@thewilkybarkid I think it's unavoidable to redeclare the properties from the `Function.prototype` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/prototype)

If Typescript did override, then it would have a knock on effect on things like objects:
```typescript
type A = {
 [property: string]: SomeType;
};

const t: A = { };

t.toString(); // is this a function or `SomeType`
```

Reply:
Opened https://stackoverflow.com/questions/59560965/name-property-when-a-type-is-both-indexable-and-callable